examples/todo-list/flutter-todo-list/lib/screens/todo.dart

```diff
@@ -0,0 +1,206 @@
+import 'package:flutter/material.dart';
+import 'package:todosupabase/constant.dart';
+import 'package:todosupabase/functions/auth.dart';
+import 'package:todosupabase/functions/crud.dart';
+
+class TodoPage extends StatefulWidget {
+  const TodoPage({Key? key}) : super(key: key);
+
+  @override
+  State<TodoPage> createState() => _TodoPageState();
+}
+
+class _TodoPageState extends State<TodoPage> {
+  String userName = "";
+  bool loading = true;
+  List<dynamic> todos = [];
+  final TextEditingController _taskEditingController = TextEditingController();
+  final TextEditingController _dueEditingController = TextEditingController();
+
+  void initFunc() async {
+    var nameVal = await CrudSupabase.getUser(
+        email: AuthSupabase.client.auth.currentUser!.email.toString());
+    setState(() {
+      userName = nameVal[0]['name'];
+    });
+    List<dynamic> value = await CrudSupabase.getTodo();
+    value.forEach((element) {
+      todos.add(element);
+    });
+    setState(() {
+      loading = false;
+    });
+  }
+
+  @override
+  void initState() {
+    // TODO: implement initState
+    super.initState();
+    initFunc();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _taskEditingController.dispose();
+    _dueEditingController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      floatingActionButton: FloatingActionButton(
+        backgroundColor: const Color(0xff33b27b),
+        onPressed: () {
+          displayTextInputDialog(context, () async {
+            await CrudSupabase.addTodo(
+              task: _taskEditingController.text,
+              due: _dueEditingController.text,
+            );
+            Navigator.pop(context);
+            setState(() {
+              loading = true;
+            });
+
+            List<dynamic> value = await CrudSupabase.getTodo();
+            todos.clear();
+            setState(() {});
+            value.forEach((element) {
+              todos.add(element);
+            });
+            setState(() {
+              loading = false;
+            });
+          }, _taskEditingController, _dueEditingController);
+        },
+        child: const Icon(Icons.add),
+      ),
+      appBar: AppBar(
+        backgroundColor: Colors.black,
+        automaticallyImplyLeading: false,
+        title: Text('Hey ${userName.split(" ")[0]}, add todo'),
+        actions: <Widget>[
+          IconButton(
+            icon: const Icon(Icons.exit_to_app),
+            onPressed: () {
+              AuthSupabase.logOut(context);
+            },
+          ),
+        ],
+      ),
+      body: loading
+          ? const Center(child: CircularProgressIndicator())
+          : SizedBox(
+              height: double.infinity,
```

Comment:
This is another `SizedBox` that we can get rid of!

Reply:
Here too if I don't use sizedbox the listview is going to give me error it require some size